packages/medusa/src/services/product.ts

```diff
@@ -609,12 +609,14 @@ class ProductService extends TransactionBaseService {
 
       const result = await productRepo.save(product)
 
-      await this.eventBus_
-        .withTransaction(manager)
-        .emit(ProductService.Events.UPDATED, {
+      await this.eventBus_.withTransaction(manager).emit(
```

Comment:
Todo: no need to use withTransaction no?

Reply:
Changed to `uniqueCacheKey`